CRISPResso2/CRISPRessoShared.py

```diff
@@ -1661,3 +1673,462 @@ def zip_results(results_folder):
+        """Safety check, if total is below minimum send warnings
+
+        Parameters:
+        -----------
+        total : int
+            The total reads, unaligned and aligned
+        """
+        if total < self.minimum:
+            self.messageHandler.display_warning(self.message)
+            self.messageHandler.report_warning(self.message)
+
+
+class OverallReadsAlignedGuardRail:
+    """Guardrail class: check if enough reads are aligned"""
+    def __init__(self, messageHandler, cutoff):
+        """Assign variables and create guardrail message
+
+        Parameters:
+        -----------
+        messageHandler : GuardRailMessagehandler
+            Guardrail message handler to create and display warnings
+        cutoff : float
+            The float representation of percentage of minimum reads to be aligned
+        """
+        self.messageHandler = messageHandler
+        self.message = " <={val}% of reads were aligned".format(val=(cutoff * 100))
+        self.cutoff = cutoff
+    
+    def safety(self, total_reads, n_read_aligned):
+        """Safety check, if total_reads divided by n_reads_aligned is lower than the cutoff
+
+        Parameters:
+        -----------
+        total : int
+            Total reads, unaligned and aligned
+        n_read_aligned : int
+            Total aligned reads
+        """
+        if total_reads == 0:
+            return
+        if (n_read_aligned/total_reads) <= self.cutoff:
+            self.messageHandler.display_warning(self.message)
+            self.messageHandler.report_warning(self.message)
+
+
+class LowReadsAlignedToAmpliconGuardRail:
+    """Guardrail class: check if the distribution of reads is roughly even across amplicons"""
+    def __init__(self, messageHandler, cutoff):
+        """Assign variables and create guardrail message
+
+        Parameters:
+        -----------
+        messageHandler : GuardRailMessagehandler
+            Guardrail message handler to create and display warnings
+        cutoff : float
+            The float representation of the accepted percentage deviation down from the expected distribution
+        """
+        self.messageHandler = messageHandler
+        self.message = " <={val}% of expected reads were aligned to amplicon: ".format(val=(cutoff * 100))
+        self.cutoff = cutoff
+
+    def safety(self, total_reads, amplicons, reads_aln_amplicon):
+        """Safety check, if total_reads divided by n_reads_aligned is lower than the total_reads divided by the number of amplicons
+        
+        Parameters:
+        -----------
+        total_reads : int
+            Total reads, unaligned and aligned
+        amplicons : list
+            A list of the names of the amplicons
+        reads_aln_amplicon : dict
+            A dictionary with the names of amplicons as the key and the number of reads aligned as the value
+        """
+        expected_per_amplicon = total_reads / len(amplicons)
+        for amplicon in amplicons:
```

Comment:
If you have the names of the amplicons as the keys in `reads_aln_amplicon` do you need to pass `amplicons` in?